File: CS50/pset1/cash.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>


int main()
{
    float num;
    do
    {
        num = get_float("Change owed: ");
    } 
    while (num < 0);
    
    int int_num = round(num * 100);  /* reduce from decimal */
  
    
    int quarter = int_num / 25;
    int left_quarter = int_num % 25;  /* get remainder */
    
    int dime = left_quarter / 10;
    int left_dime = left_quarter % 10;
    
    int nickel = left_dime / 5;
    int left_nickel = left_dime % 5;
    
    int penny = left_nickel;
    
    /* control situation:
    printf("quarter: %d\n", quarter);
    printf("left_quarter: %d\n", left_quarter);
    
    printf("dime: %d\n", dime);
    printf("left_dime: %d\n", left_dime);
    
    printf("nickel: %d\n", nickel);
    printf("left_nickel: %d\n", left_nickel);
    
    printf("penny: %d\n", penny); */
    
    int total = quarter + dime + nickel + penny;
    
    printf("Total: %d\n", total);
    
    return 0;
}

File: CS50/pset6/cash.py
import cs50
# cs50.getfloat()

while True:
    money = cs50.get_float("Change owed: ")
    if money > 0:
        break

t_money = money * 100

quarter = 25
dime = 10
nickel = 5
penny = 1

c_quarter = 0
c_dime = 0
c_nickel = 0
c_penny = 0

c_quarter = t_money // quarter
t_money = t_money % quarter

c_dime = t_money // dime
t_money = t_money % dime

c_nickel = t_money // nickel
t_money = t_money % nickel

c_penny = t_money // penny
t_money = t_money % penny

answer = int(c_quarter + c_dime + c_nickel + c_penny)

print(answer)